src/main/gov/nasa/jpf/jdart/constraints/ConstraintsTree.java

```diff
@@ -107,6 +109,50 @@ private String toString(final String prefix, boolean values, boolean postconditi
                     
       return ret;
     }
+
+    public void toJson() {
+      toJson(false, true);
+    }
+
+    private void toJson(boolean values, boolean postconditions) {
+      String tree = toJson("", "", values, postconditions);
+      
+      try(PrintWriter out = new PrintWriter("tree.json")){
+        out.println(tree);
+        out.close();
+      } catch (java.io.FileNotFoundException e) {
+        System.out.println("Cannot write json file.");
+      }
+    }
+
+    private String toJson(String prefix, String branch, boolean values, boolean postconditions) {
+      String oriPrefix = prefix;
```

Comment:
This method of outputting to json is **really** inefficient. Strings are immutable in java, so whenever you concatenate strings, you have to create a new one a copy the two strings into it. That is okay for simple output, but when you are iteratively doing this, it becomes extremely expensive.
I would strongly suggest to use a stringbuilder. Even better, use the `google gson` library (which is included in jdart---see the lib folder) for json output. With that, you can probably reduce this code to a few lines.